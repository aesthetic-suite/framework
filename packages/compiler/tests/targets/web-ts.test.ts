import runTargetTests from './runTargetTests';

runTargetTests('web', 'web-ts', true);
