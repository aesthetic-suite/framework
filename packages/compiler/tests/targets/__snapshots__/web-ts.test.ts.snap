// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Format web-ts compiles and writes files 1`] = `
"// Auto-generated by Aesthetic. Do not modify!
import {
  BorderTokens,
  BreakpointTokens,
  DepthTokens,
  HeadingTokens,
  ShadowTokens,
  SpacingTokens,
  TextTokens,
  TypographyToken,
} from '@aesthetic/system';

export const border: BorderTokens = {
  sm: {
    radius: '0.11rem', // 1.50px
    width: '0rem', // 0px
  },
  df: {
    radius: '0.21rem', // 3px
    width: '0.07rem', // 1px
  },
  lg: {
    radius: '0.32rem', // 4.50px
    width: '0.14rem', // 2px
  },
};

export const breakpoint: BreakpointTokens = {
  xs: {
    query: 'screen and (min-width: 45.71em)',
    querySize: 640,
    rootLineHeight: 1.33,
    rootTextSize: '14.94px',
  },
  sm: {
    query: 'screen and (min-width: 68.57em)',
    querySize: 960,
    rootLineHeight: 1.42,
    rootTextSize: '15.94px',
  },
  md: {
    query: 'screen and (min-width: 91.43em)',
    querySize: 1280,
    rootLineHeight: 1.52,
    rootTextSize: '17.01px',
  },
  lg: {
    query: 'screen and (min-width: 114.29em)',
    querySize: 1600,
    rootLineHeight: 1.62,
    rootTextSize: '18.15px',
  },
  xl: {
    query: 'screen and (min-width: 137.14em)',
    querySize: 1920,
    rootLineHeight: 1.73,
    rootTextSize: '19.36px',
  },
};

export const depth: DepthTokens = {
  content: 100,
  navigation: 1000,
  sheet: 1100,
  overlay: 1200,
  modal: 1300,
  toast: 1400,
  dialog: 1500,
  menu: 1600,
  tooltip: 1700,
};

export const heading: HeadingTokens = {
  l1: {
    letterSpacing: '0.25px',
    lineHeight: 1.5,
    size: '1.14rem', // 16px
  },
  l2: {
    letterSpacing: '0.33px',
    lineHeight: 1.69,
    size: '1.43rem', // 20px
  },
  l3: {
    letterSpacing: '0.44px',
    lineHeight: 1.9,
    size: '1.79rem', // 25px
  },
  l4: {
    letterSpacing: '0.59px',
    lineHeight: 2.14,
    size: '2.23rem', // 31.25px
  },
  l5: {
    letterSpacing: '0.79px',
    lineHeight: 2.4,
    size: '2.79rem', // 39px
  },
  l6: {
    letterSpacing: '1.05px',
    lineHeight: 2.7,
    size: '3.49rem', // 48.80px
  },
};

export const shadow: ShadowTokens = {
  xs: {
    x: '0rem', // 0px
    y: '0.14rem', // 2px
    blur: '0.14rem', // 2px
    spread: '0rem', // 0px
  },
  sm: {
    x: '0rem', // 0px
    y: '0.23rem', // 3.24px
    blur: '0.27rem', // 3.75px
    spread: '0rem', // 0px
  },
  md: {
    x: '0rem', // 0px
    y: '0.37rem', // 5.24px
    blur: '0.39rem', // 5.50px
    spread: '0rem', // 0px
  },
  lg: {
    x: '0rem', // 0px
    y: '0.61rem', // 8.47px
    blur: '0.52rem', // 7.25px
    spread: '0rem', // 0px
  },
  xl: {
    x: '0rem', // 0px
    y: '0.98rem', // 13.71px
    blur: '0.64rem', // 9px
    spread: '0rem', // 0px
  },
};

export const spacing: SpacingTokens = {
  xs: '0.31rem', // 4.38px
  sm: '0.63rem', // 8.75px
  df: '1.25rem', // 17.50px
  md: '2.50rem', // 35px
  lg: '3.75rem', // 52.50px
  xl: '5rem', // 70px
  type: 'vertical-rhythm',
  unit: 17.5,
};

export const text: TextTokens = {
  sm: {
    lineHeight: 1.25,
    size: '0.89rem', // 12.45px
  },
  df: {
    lineHeight: 1.25,
    size: '1rem', // 14px
  },
  lg: {
    lineHeight: 1.25,
    size: '1.13rem', // 15.75px
  },
};

export const typography: TypographyToken = {
  font: {
    heading:
      'system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, \\"Helvetica Neue\\", Arial, \\"Noto Sans\\", sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\", \\"Noto Color Emoji\\"',
    locale: {},
    monospace:
      'Menlo, Monaco, Consolas, \\"Lucida Console\\", \\"Liberation Mono\\", \\"Courier New\\", monospace',
    text:
      'system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, \\"Helvetica Neue\\", Arial, \\"Noto Sans\\", sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\", \\"Noto Color Emoji\\"',
    system:
      'system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, \\"Helvetica Neue\\", Arial, \\"Noto Sans\\", sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\", \\"Noto Color Emoji\\"',
  },
  rootLineHeight: 1.25,
  rootTextSize: '14px',
};
"
`;

exports[`Format web-ts compiles and writes files 2`] = `
"// Auto-generated by Aesthetic. Do not modify!
import { PaletteToken } from '@aesthetic/system';

// contrast: normal
// scheme: light

export * from '..';

export const brand: PaletteToken = {
  color: {
    '00': '#000',
    '10': '#000',
    '20': '#000',
    '30': '#000',
    '40': '#000',
    '50': '#000',
    '60': '#000',
    '70': '#000',
    '80': '#000',
    '90': '#000',
  },
  bg: {
    base: '#000',
    focused: '#000',
    hovered: '#000',
    selected: '#000',
    disabled: '#000',
  },
  fg: {
    base: '#000',
    focused: '#000',
    hovered: '#000',
    selected: '#000',
    disabled: '#000',
  },
  text: '#000',
};

export const primary: PaletteToken = {
  color: {
    '00': '#000',
    '10': '#000',
    '20': '#000',
    '30': '#000',
    '40': '#000',
    '50': '#000',
    '60': '#000',
    '70': '#000',
    '80': '#000',
    '90': '#000',
  },
  bg: {
    base: '#000',
    focused: '#000',
    hovered: '#000',
    selected: '#000',
    disabled: '#000',
  },
  fg: {
    base: '#000',
    focused: '#000',
    hovered: '#000',
    selected: '#000',
    disabled: '#000',
  },
  text: '#000',
};

export const secondary: PaletteToken = {
  color: {
    '00': '#000',
    '10': '#000',
    '20': '#000',
    '30': '#000',
    '40': '#000',
    '50': '#000',
    '60': '#000',
    '70': '#000',
    '80': '#000',
    '90': '#000',
  },
  bg: {
    base: '#000',
    focused: '#000',
    hovered: '#000',
    selected: '#000',
    disabled: '#000',
  },
  fg: {
    base: '#000',
    focused: '#000',
    hovered: '#000',
    selected: '#000',
    disabled: '#000',
  },
  text: '#000',
};

export const tertiary: PaletteToken = {
  color: {
    '00': '#000',
    '10': '#000',
    '20': '#000',
    '30': '#000',
    '40': '#000',
    '50': '#000',
    '60': '#000',
    '70': '#000',
    '80': '#000',
    '90': '#000',
  },
  bg: {
    base: '#000',
    focused: '#000',
    hovered: '#000',
    selected: '#000',
    disabled: '#000',
  },
  fg: {
    base: '#000',
    focused: '#000',
    hovered: '#000',
    selected: '#000',
    disabled: '#000',
  },
  text: '#000',
};

export const neutral: PaletteToken = {
  color: {
    '00': '#000',
    '10': '#000',
    '20': '#000',
    '30': '#000',
    '40': '#000',
    '50': '#000',
    '60': '#000',
    '70': '#000',
    '80': '#000',
    '90': '#000',
  },
  bg: {
    base: '#000',
    focused: '#000',
    hovered: '#000',
    selected: '#000',
    disabled: '#000',
  },
  fg: {
    base: '#000',
    focused: '#000',
    hovered: '#000',
    selected: '#000',
    disabled: '#000',
  },
  text: '#000',
};

export const muted: PaletteToken = {
  color: {
    '00': '#000',
    '10': '#000',
    '20': '#000',
    '30': '#000',
    '40': '#000',
    '50': '#000',
    '60': '#000',
    '70': '#000',
    '80': '#000',
    '90': '#000',
  },
  bg: {
    base: '#000',
    focused: '#000',
    hovered: '#000',
    selected: '#000',
    disabled: '#000',
  },
  fg: {
    base: '#000',
    focused: '#000',
    hovered: '#000',
    selected: '#000',
    disabled: '#000',
  },
  text: '#000',
};

export const info: PaletteToken = {
  color: {
    '00': '#000',
    '10': '#000',
    '20': '#000',
    '30': '#000',
    '40': '#000',
    '50': '#000',
    '60': '#000',
    '70': '#000',
    '80': '#000',
    '90': '#000',
  },
  bg: {
    base: '#000',
    focused: '#000',
    hovered: '#000',
    selected: '#000',
    disabled: '#000',
  },
  fg: {
    base: '#000',
    focused: '#000',
    hovered: '#000',
    selected: '#000',
    disabled: '#000',
  },
  text: '#000',
};

export const warning: PaletteToken = {
  color: {
    '00': '#000',
    '10': '#000',
    '20': '#000',
    '30': '#000',
    '40': '#000',
    '50': '#000',
    '60': '#000',
    '70': '#000',
    '80': '#000',
    '90': '#000',
  },
  bg: {
    base: '#000',
    focused: '#000',
    hovered: '#000',
    selected: '#000',
    disabled: '#000',
  },
  fg: {
    base: '#000',
    focused: '#000',
    hovered: '#000',
    selected: '#000',
    disabled: '#000',
  },
  text: '#000',
};

export const danger: PaletteToken = {
  color: {
    '00': '#000',
    '10': '#000',
    '20': '#000',
    '30': '#000',
    '40': '#000',
    '50': '#000',
    '60': '#000',
    '70': '#000',
    '80': '#000',
    '90': '#000',
  },
  bg: {
    base: '#000',
    focused: '#000',
    hovered: '#000',
    selected: '#000',
    disabled: '#000',
  },
  fg: {
    base: '#000',
    focused: '#000',
    hovered: '#000',
    selected: '#000',
    disabled: '#000',
  },
  text: '#000',
};

export const failure: PaletteToken = {
  color: {
    '00': '#000',
    '10': '#000',
    '20': '#000',
    '30': '#000',
    '40': '#000',
    '50': '#000',
    '60': '#000',
    '70': '#000',
    '80': '#000',
    '90': '#000',
  },
  bg: {
    base: '#000',
    focused: '#000',
    hovered: '#000',
    selected: '#000',
    disabled: '#000',
  },
  fg: {
    base: '#000',
    focused: '#000',
    hovered: '#000',
    selected: '#000',
    disabled: '#000',
  },
  text: '#000',
};

export const success: PaletteToken = {
  color: {
    '00': '#000',
    '10': '#000',
    '20': '#000',
    '30': '#000',
    '40': '#000',
    '50': '#000',
    '60': '#000',
    '70': '#000',
    '80': '#000',
    '90': '#000',
  },
  bg: {
    base: '#000',
    focused: '#000',
    hovered: '#000',
    selected: '#000',
    disabled: '#000',
  },
  fg: {
    base: '#000',
    focused: '#000',
    hovered: '#000',
    selected: '#000',
    disabled: '#000',
  },
  text: '#000',
};
"
`;
